nemoguardrails/embeddings/basic.py

```
@@ -187,70 +190,85 @@ async def build(self):
-        # Compute the embeddings
-        embeddings = await self._get_embeddings(batch)
-        for i in range(len(embeddings)):
-            self._req_results[batch_ids[i]] = embeddings[i]
-
-        # Signal that the batch has finished processing
-        batch_event.set()
+        try:
+            embeddings = await self._get_embeddings(batch)
+            for i in range(len(embeddings)):
+                self._req_results[batch_ids[i]] = embeddings[i]
+        except BaseException as exc:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `BaseException`

(BLE001)